receiver/bindplaneauditlogs/logs.go

```
@@ -0,0 +1,232 @@
+}
+
+type bindplaneAuditLogsReceiver struct {
+	cfg           Config
+	client        httpClient
+	consumer      consumer.Logs
+	logger        *zap.Logger
+	cancel        context.CancelFunc
+	wg            *sync.WaitGroup
+	lastTimestamp time.Time
+}
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+	CloseIdleConnections()
+}
+
+// newBindplaneAuditLogsReceiver returns a newly configured bindplaneAuditLogsReceiver
+func newBindplaneAuditLogsReceiver(cfg *Config, logger *zap.Logger, consumer consumer.Logs) (*bindplaneAuditLogsReceiver, error) {
+	return &bindplaneAuditLogsReceiver{
+		cfg:           *cfg,
+		client:        http.DefaultClient,
+		consumer:      consumer,
+		logger:        logger,
+		wg:            &sync.WaitGroup{},
+		lastTimestamp: time.Now().UTC().Add(-cfg.PollInterval),
+	}, nil
+}
+
+func (r *bindplaneAuditLogsReceiver) Start(_ context.Context, _ component.Host) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	r.cancel = cancel
+	r.wg.Add(1)
+	go r.startPolling(ctx)
+	return nil
+}
+
+func (r *bindplaneAuditLogsReceiver) startPolling(ctx context.Context) {
+	defer r.wg.Done()
+	t := time.NewTicker(r.cfg.PollInterval)
+
+	err := r.poll(ctx)
+	if err != nil {
+		r.logger.Error("there was an error during the first poll", zap.Error(err))
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			err := r.poll(ctx)
+			if err != nil {
+				r.logger.Error("there was an error during the poll", zap.Error(err))
+			}
+		}
+	}
+}
+
+func (r *bindplaneAuditLogsReceiver) poll(ctx context.Context) error {
+	logEvents := r.getLogs(ctx)
+	observedTime := pcommon.NewTimestampFromTime(time.Now())
+	logs := r.processLogEvents(observedTime, logEvents)
+	if logs.LogRecordCount() > 0 {
+		if err := r.consumer.ConsumeLogs(ctx, logs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *bindplaneAuditLogsReceiver) getLogs(ctx context.Context) []AuditLogEvent {
```

This function should get reworked to return an error instead of logging an error. This way we can check if there was an error inside of `poll` before processing. If there was then we can return the error and `startPolling` will catch it and log the error message.